backend/app/store/search/bleve_engine.go

```diff
@@ -0,0 +1,207 @@
+package search
+
+import (
+	"fmt"
+	"github.com/blevesearch/bleve/v2"
+	bleveCustom "github.com/blevesearch/bleve/v2/analysis/analyzer/custom"
+	bleveStandard "github.com/blevesearch/bleve/v2/analysis/analyzer/standard"
+	bleveEn "github.com/blevesearch/bleve/v2/analysis/lang/en"
+	bleveRu "github.com/blevesearch/bleve/v2/analysis/lang/ru"
+	"github.com/blevesearch/bleve/v2/analysis/token/lowercase"
+	bleveSingle "github.com/blevesearch/bleve/v2/analysis/tokenizer/single"
+	"github.com/blevesearch/bleve/v2/mapping"
+	"github.com/pkg/errors"
+	"github.com/umputun/remark42/backend/app/store"
+	"log"
+	"os"
+)
+
+const userFieldName = "user"
+const textFieldName = "text"
+
+type bleveEngine struct {
+	index     bleve.Index
+	indexPath string
+}
+
+// Available text analyzers.
+// Bleve supports a bit more languages that may be added,
+// see https://github.com/blevesearch/bleve/tree/master/analysis/lang
+var analyzerMapping = map[string]string{
+	"standard": bleveStandard.Name,
+	"english":  bleveEn.AnalyzerName,
+	"russian":  bleveRu.AnalyzerName,
+}
+
+func newBleveEngine(indexPath, analyzer string) (*bleveEngine, error) {
+	if _, ok := analyzerMapping[analyzer]; !ok {
+		analyzers := make([]string, 0, len(analyzerMapping))
+		for k := range analyzerMapping {
+			analyzers = append(analyzers, k)
+		}
+		return nil, errors.Errorf("Unknown analyzer: %q. Available analyzers for bleve: %v", analyzer, analyzers)
+	}
+	var index bleve.Index
+	var err error
+
+	st, errOpen := os.Stat(indexPath)
+	switch {
+	case os.IsNotExist(errOpen):
+		// create new
+		log.Printf("[INFO] creating new search index %s", indexPath)
+		index, err = bleve.New(indexPath, createIndexMapping(analyzerMapping[analyzer]))
+		if err != nil {
+			return nil, errors.Wrap(err, "cannot open index")
+		}
+	case errOpen == nil:
+		// open existing
+		if !st.IsDir() {
+			return nil, errors.Errorf("index path should be a directory")
+		}
+		log.Printf("[INFO] opening existing search index %s", indexPath)
+		index, err = bleve.Open(indexPath)
+		if err != nil {
+			return nil, errors.Wrap(err, "cannot create index")
+		}
+	default:
+		// error
+		return nil, errors.Wrap(err, "cannot open index")
+	}
+
+	eng := &bleveEngine{
+		index:     index,
+		indexPath: indexPath,
+	}
+
+	return eng, nil
+}
+
+// Index documents
+func (b *bleveEngine) Index(comments []*store.Comment) error {
+	batch := b.index.NewBatch()
+
+	for _, comment := range comments {
+		err := batch.Index(comment.ID, comment)
```

Comment:
Not sure that we should index `store.Comment` directly, maybe it's better to convert it to simpler plain structure and store it to index.

It will simplify bleve mapping for struct (see `createIndexMapping`), and we won't need to specify `user.name` (instead of `user` or `username` if we put it on top level of struct) when we query for username.

Like it was in first version
https://github.com/umputun/remark42/blob/d755ebaca2361216f06813d8f8b6d5d953f4bdc4/backend/app/store/search/types/types.go#L72-L82